sample_information_validator.py

```diff
@@ -23,78 +23,280 @@
+            else:
+                cellID_noSample.append(cell_id)# TODO check here that these rows do really not contain information
+        return(cellID_withSample)
+
+    def ProjectInfo(self, username_couchDB, password_couchDB, url):
+        url_string = "http://"+username_couchDB+":"+password_couchDB+url
+        connection = couchdb.Server(url=url_string)
+        db = connection["projects"]
+        # check the existence of the project number in couchDB
+        project_id_found = db.view("project/project_id", key=self.projectID())
+        prow=project_id_found.rows
+        # Project not found
+        if len(prow) == 0:
             logger.error(
-                    'Cell {} is not numeric'.format(cell.coordinate)
-                    )
-            return False
-        try:
-            float(cell.value)
-            return True
-        except ValueError:
+            'Project not found, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
+            )
+            quit()
+        # more than one project found
+        elif len(prow) > 1:
             logger.error(
-                    'Cell {} is numeric but cannot be '
-                    'transformed to float'.format(cell.coordinate)
+            'Project ID not unique, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
             )
-            return False
-        except TypeError:
-            if cell.value is None:
-                logger.warning(
-                    'Cell {} is numeric but empty'.format(cell.coordinate)
+            quit()
+        else:
+        # puts the Document of the identified project in a new variable "pdoc"
+            pdoc = db.get(prow[0].id)
+            return(pdoc)
+
+    def requirements(self, path):
+        if self.sample_rec == None:
+            wb_sample_recommendations = load_workbook(path, read_only=True, data_only=True)
+            self.sample_rec = wb_sample_recommendations['Numeric']
+
+    def prep_standards(self, info, recom_path):
+        ''' gets the sample requirements from the sample requirement excel sheet based
+            on the given sample prep type. '''
+
+        prep = info['details']['library_construction_method']
+
+        prep_recs = [None,None,None,None,None,None,None,None]
+        prep_type_found = False
+        for row in range(2, 15):
+            cellID_prep = "A{row}".format(row=row)
+            self.requirements(recom_path)
+            if(self.sample_rec[cellID_prep].value == prep):
+                prep_row = row
+                prep_recs = [\
+                self.sample_rec["C{min_conc}".format(min_conc=prep_row)].value, \
+                self.sample_rec["D{max_conc}".format(max_conc=prep_row)].value, \
```

Comment:
I don't understand these lines. `prep_row` should be the same every time right? How can it then be both min_conc, max_conc and so on?

Reply:
it is the same value, I just called it min_conc and max_conc for readability. The true difference comes from "C" and "D". BUT I realized that that is still hard coded and shouldn't be, they are already class variables, so it's not biggie to change.

Reply:
Ah, I see now! I can't find exactly those class variables though. Maybe add the column as variables and use `prep_row` as a variable? Perhaps simplify it to: `self.sample_rec["{0}{1}".format(min_conc_col, prep_row)].value`?